crates/biome_js_analyze/src/analyzers/nursery/no_useless_lone_block_statements.rs

```diff
@@ -0,0 +1,125 @@
+use biome_js_syntax::JsVariableKind::{Const, Let};
+use biome_rowan::{AstNode, AstNodeList, TextRange};
+
+declare_rule! {
+    /// Disallow unnecessary nested block statements.
+    ///
+    /// > In JavaScript, prior to ES6, standalone code blocks delimited by curly braces do not create a new scope and have no use.
+    /// > In ES6, code blocks may create a new scope if a block-level binding (let and const), a class declaration or a function declaration (in strict mode) are present. A block is not considered redundant in these cases.
+    ///
+    /// Source: https://eslint.org/docs/latest/rules/no-lone-blocks
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// {}
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// if (foo) {
+    ///   bar();
+    ///   {
+    ///     baz();
+    ///   }
+    /// }
+    /// ```
+    ///
+    /// ## Valid
+    ///
+    /// ```js
+    /// while (foo) {
+    ///   bar();
+    /// }
+    /// ```
+    ///
+    pub(crate) NoUselessLoneBlockStatements {
+        version: "next",
+        name: "noUselessLoneBlockStatements",
+        recommended: false,
+    }
+}
+
+impl Rule for NoUselessLoneBlockStatements {
+    type Query = Semantic<JsBlockStatement>;
+    type State = TextRange;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let block = ctx.query();
+
+        if let Some(_labeled_statement) = JsLabeledStatement::cast(block.syntax().parent()?) {
+            return None;
+        }
+
+        if block.statements().is_empty() {
+            return Some(block.range());
+        }
+
+        if block_has_control_structure(&block) {
+            return None;
+        }
+
+        for statement in block.statements() {
+            return match statement {
+                AnyJsStatement::JsVariableStatement(variable) if is_let_or_const_statement(&variable) => None,
+                AnyJsStatement::JsClassDeclaration(_) => None,
+                AnyJsStatement::JsFunctionDeclaration(func) if is_block_scoped_function(&func) => None,
+                _ => Some(block.range()),
+            }
+        }
+
+        None
+    }
+
+    fn diagnostic(_: &RuleContext<Self>, range: &Self::State) -> Option<RuleDiagnostic> {
+        Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                range,
+                markup! {
+                    "This block statement doesn't serve any purpose and can be safely removed."
+                },
+            )
+                .note(markup! {
+                "Standalone block statements without any block-level declarations are redundant in JavaScript and can be removed to simplify the code."
+            }),
+        )
+    }
+}
+
+fn block_has_control_structure(block: &JsBlockStatement) -> bool {
```

Comment:
What about this name instead?
```suggestion
fn in_control_structure(block: &JsBlockStatement) -> bool {
```